core/src/main/java/com/linecorp/armeria/server/http/dynamic/DynamicHttpFunction.java

```diff
@@ -0,0 +1,18 @@
+package com.linecorp.armeria.server.http.dynamic;
+
+import java.util.Map;
+
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * Path variable-enabled function.
+ *
+ * <p>Mapped path variables are passed via arguments parameter.
+ */
+@FunctionalInterface
+public interface DynamicHttpFunction {
+    HttpResponse serve(ServiceRequestContext ctx, HttpRequest req, Map<String, String> bound)
```

Comment:
- bound -> params or args
- How about adding `HttpResponseWriter` parameter to this method so that a user does not have to create a new `DefaultHttpResponse` by him/herself.